runatlantis.io/docs/security.md

```diff
@@ -47,6 +50,19 @@ For example:
 
 This flag ensures your Atlantis install isn't being used with repositories you don't control. See `atlantis server --help` for more details.
 
+### Protect Terraform Planning
+If attackers submitting pull requests with malicious Terraform code is in your threat model
+then you must be aware that `terraform apply` approvals are not enough. It is possible
+to run malicious code in a `terraform plan` using the [`external` data source](https://registry.terraform.io/providers/hashicorp/external/latest/docs/data-sources/data_source)
+or by specifying a malicious provider. This code could then exfiltrate your credentials.
+
+To prevent this, you could:
+1. Bake providers into the Atlantis image or host and deny egress in production.
+1. Implement the provider registry protocol internally and deny public egress, that way you control who has write access to the registry.
```

Comment:
@nishkrishnan please help with this section

Reply:
It's worth noting that external data source would likely not be available for use if you don't add it to your internal provider registry. This is an assumption though.